app/models/manageiq/providers/oracle/infra_manager.rb

```diff
@@ -0,0 +1,98 @@
+class ManageIQ::Providers::Oracle::InfraManager < ManageIQ::Providers::InfraManager
+  require_nested :RefreshWorker
+  require_nested :RefreshParser
+  require_nested :Host
+  require_nested :Refresher
+  require_nested :Template
+  require_nested :Vm
+
+  def self.ems_type
+    @ems_type ||= "oraclevm".freeze
+  end
+
+  def self.description
+    @description ||= "Oracle Virtualization Manager".freeze
+  end
+
+  def self.default_blacklisted_event_names
+    []
+  end
+
+  def supports_port?
+    true
+  end
+
+  def supported_auth_types
+    %w(default)
+  end
+
+  def supports_authentication?(authtype)
+    supported_auth_types.include?(authtype.to_s)
+  end
+
+  def self.raw_connect(server, port, username, password)
+    require 'fog/oracle'
+
+    url = "https://#{server}:#{port}"
```

Comment:
Thanks, it's been a while, looks like you have to convert it to a string before you connect: 
https://github.com/fog/fog-core/blob/f6bf9c0d34040085d7b962935a397bd9df5db74b/lib/fog/core/connection.rb#L31

So, `URI::HTTPS.build(:host => server, :port => port).to_s`